gamedata/alldefs_post.lua

```diff
@@ -1454,31 +1454,30 @@ function WeaponDef_Post(name, wDef)
 		end
 
 		if modOptions.shieldsrework == true then
-			local shieldCollisionExemptions = { --add the name of the weapons (or just the name of the unit followed by _ ) to this table to exempt from shield collision. 
-				'corsilo_', 'armsilo_', 'armthor_empmissile', 'armemp_', 'cortron_', 'corjuno_', 'armjuno_'
-			}
-
 			-- For balance, paralyzers need to do reduced damage to shields, as their raw raw damage is outsized
 			local paralyzerShieldDamageMultiplier = 0.25
 
-			if wDef.shield then
-				wDef.shield.repulser = false
-				wDef.shield.exterior = true
-			end
+			-- VTOL's may or may not do full damage to shields if not defined in weapondefs
+			local vtolShieldDamageMultiplier = 0
 
-			wDef.interceptedbyshieldtype = 1
+			local shieldCollisionExemptions = { --add the name of the weapons (or just the name of the unit followed by _ ) to this table to exempt from shield collision. 
+			'corsilo_', 'armsilo_', 'armthor_empmissile', 'armemp_', 'cortron_', 'corjuno_', 'armjuno_'
+		}
 
-			for _, exemption in ipairs(shieldCollisionExemptions) do
-				if string.find(name, exemption)then
-					wDef.interceptedbyshieldtype = 2
-					break
+			if wDef.damage ~= nil then
+				-- Due to the engine not handling overkill damage, we have to store the original shield damage values as a customParam for unit_shield_behavior.lua to reference
+				wDef.customparams = wDef.customparams or {}
+				if wDef.damage.shields then
+					wDef.customparams.shield_damage = wDef.damage.shields
+				elseif wDef.damage.default then
+					wDef.customparams.shield_damage = wDef.damage.default
+				elseif wDef.damage.vtol then
+					wDef.customparams.shield_damage = wDef.damage.vtol * vtolShieldDamageMultiplier
```

Comment:
This is not a good way to deal with weapons having no defined `shields` or `default` damage entries. This is fine as a stopgap measure, but the balance team will have to add `shields` damages entries for all the weapons that need them.

Reply:
Stopgap it is. Agreed.

It also will catch people not defining a shield or default damage which implies to me they don't want it to damage shields.